lib/Model.js

```diff
@@ -0,0 +1,88 @@
+// ## Model
+
+// Will require the `Data Store` module, and assign it's `store` property to a temporary variable.
+
+// Define a constructor that accepts a singl argument `schema` and assign it's value to a public property named `schema`.
+
+var db = require('./DataStore.js').store;
+
+function Model (schema){
+  this.schema = schema;
+  this.id = null;
+
+  for (var key in schema){
+    this[key] = schema[key];
+  }
+
+  if (!db.hasOwnProperty(this.constructor.name)){
+    db[this.constructor.name] = [];
+  } 
+}
+
+//prototype - save & destroy methods
+
+Model.prototype.save = function(){
+  if (!this.id){
+    var nextId = this.constructor.getNextId();
+    this.id = nextId;
+  }
+  db[this.constructor.name].push(this);
```

Comment:
So although this works, we're actually saving this instance to the db. That means that when changes are made to this instance, the changes will be reflected in the version in the db. This is not what we want because we may want to store different versions of this instance rather than just pushing the same instance on to this array over and over again.